### PR TITLE
fix: show user modal if no longer in the room

### PR DESCRIPTION
**Pull request by nadirabbas**
Clicking on a user in chat doesn't open their profile if they have left the room.
This PR adds the ability to make a request to API to get user profile ONLY IF they have left the room.

### CODE REVIEW

**Comment by nadirabbas** [comment]
@benawad we have this: `Cannot find module: '@material-ui/core/styles'. Make sure this package is installed.` idk why

File: kofta/src/vscode-webview/types.ts [hunk 1]
```diff
@@ -1,58 +1,58 @@
...
+	muteMap: Record<string, boolean>;
+	activeSpeakerMap: Record<string, boolean>;
+	autoSpeaker: boolean;
 };
 
 export type WsParam = {
-  op: string;
-  d: any;
+	op: string;
+	d: any;
```
Comment (benawad) [inline kofta/src/vscode-webview/types.ts]: keep og spacing

File: kofta/src/vscode-webview/atoms.ts [hunk 2]
```diff
@@ -3,88 +3,101 @@ import { useCurrentRoomStore } from "../webrtc/stores/useCurrentRoomStore";
 import { Room, BaseUser, UserWithFollowInfo } from "./types";
 
 const createSetter = <T>(a: WritableAtom<T, any>) =>
-  atom(null, (get, set, fn: (x: T) => T) => {
-    set(a, typeof fn === "function" ? fn(get(a)) : fn);
-  });
+	atom(null, (get, set, fn: (x: T) => T) => {
```
Comment (benawad) [inline kofta/src/vscode-webview/atoms.ts]: use og spacing

(is prettier hook doing this?)

File: kofta/src/vscode-webview/modules/room-chat/ProfileModalFetcher.tsx [hunk 3]
```diff
@@ -1,53 +1,68 @@
...
-  );
+	const profileFromRoom: RoomUser | undefined = room?.users.find((x) =>
+		[x.id, x.username].includes(userId)
+	);
+
+	const profile = profileFromRoom || userProfile;
+
+	useEffect(() => {
+		profileFromRoom ||
+			wsend({ op: "get_user_profile", d: { userId, userIdType } });
```
Comment (benawad) [inline kofta/src/vscode-webview/modules/room-chat/ProfileModalFetcher.tsx]: this is a clunky pattern, I've switched to using `react-query` to fetch stuff see `PublicRoomsQuery` as an example  https://github.com/benawad/dogehouse/blob/staging/kofta/src/vscode-webview/pages/Home.tsx#L42

(you'll also use an f_handler for this on the backend)

File: kofta/src/vscode-webview/atoms.ts (hunk 2 above)
Comment (nadirabbas) [reply kofta/src/vscode-webview/atoms.ts]: @benawad its actually the pre-hook commit. you have the run `npm install` in the root to enable it for yourself too,

File: kofta/src/vscode-webview/modules/room-chat/ProfileModalFetcher.tsx (hunk 3 above)
Comment (nadirabbas) [reply kofta/src/vscode-webview/modules/room-chat/ProfileModalFetcher.tsx]: Even if I use `react-query`, I can still broadcast events to other people right?

File: kofta/src/vscode-webview/atoms.ts [hunk 4]
```diff
@@ -3,88 +3,101 @@ import { useCurrentRoomStore } from "../webrtc/stores/useCurrentRoomStore";
...
+	}
+
+	const isCreator = me.id === room.creatorId;
+
+	return {
+		isCreator,
+		isMod,
+		isSpeaker,
+		canSpeak: isCreator || isSpeaker,
+	};
```
Comment (nadirabbas) [inline kofta/src/vscode-webview/atoms.ts]: hook changed spacing on this file @benawad

**Comment by nadirabbas** [comment]
@benawad Now it's using `wsFetch`. btw all these spacing-related changes are due to the pre-commit hook. I just installed it.

File: kofta/src/vscode-webview/atoms.ts (hunk 2 above)
Comment (benawad) [reply kofta/src/vscode-webview/atoms.ts]: shouldn't the hook only format files you touch? Also can we configure it to use 2 spaces

Comment (nadirabbas) [reply kofta/src/vscode-webview/atoms.ts]: @benawad it does it project wide

File: kofta/src/vscode-webview/modules/room-chat/ProfileModalFetcher.tsx (hunk 3 above)
Comment (benawad) [reply kofta/src/vscode-webview/modules/room-chat/ProfileModalFetcher.tsx]: yeah it just replaces wsend

File: kofta/src/vscode-webview/atoms.ts (hunk 2 above)
Comment (benawad) [reply kofta/src/vscode-webview/atoms.ts]: we should move husky to `kofta`, the point of the hook is to style other contributors commits that don't know the format yet, but I doubt many of them will `npm i` at the root

**Pull request merged**